framework/include/utils/MooseArray.h

```diff
@@ -293,6 +293,7 @@ template <typename T>
 inline void
 MooseArray<T>::shallowCopy(const MooseArray & rhs)
 {
+  _data_ptr.release();
```

Comment:
No it was not causing any issue but it just seemed like the right thing to do to avoid some yet unforeseen case.

I also can't say I understand ***why*** we've never seen any issues before. I guess we've just always had some other object with access to the pointed-to data...?